client/state/login/actions.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Internal dependencies
+ */
+import {
+	LOGIN_REQUEST,
+	LOGIN_REQUEST_FAILURE,
+	LOGIN_REQUEST_SUCCESS,
+} from 'state/action-types';
+
+/**
+ * Attempt to login a user.
+ *
+ * @param  {String}    username_or_email  Username or email of the user.
+ * @param  {String}    password           Password of the user.
+ * @return {Function}                     Action thunk to trigger the login process.
+ */
+export const loginUser = ( username_or_email, password ) => {
+	return ( dispatch ) => {
+		dispatch( {
+			type: LOGIN_REQUEST,
+			username_or_email,
+			password
+		} );
+
+		// TODO: call the new login endpoint here
+		return Promise.resolve()
+			.then( ( data ) => {
+				dispatch( {
+					type: LOGIN_REQUEST_SUCCESS,
+					username_or_email,
+					password,
```

Comment:
It shouldn't be stored anywhere in the state but you're right I don't think we even need to send it as part of the action.